runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -239,6 +239,7 @@ public static Environment createProcessEnvironment(
     capabilities.addAll(ModelCoders.urns());
     capabilities.add(BeamUrns.getUrn(StandardProtocols.Enum.MULTI_CORE_BUNDLE_PROCESSING));
     capabilities.add(BeamUrns.getUrn(StandardProtocols.Enum.PROGRESS_REPORTING));
+    capabilities.add("beam:version:sdk_base:" + JAVA_SDK_HARNESS_CONTAINER_URL);
```

Comment:
What about going with `"beam:sdk:version:" + ReleaseInfo.getReleaseInfo().getSdkVersion()` and `"beam:sdk:language:java"`?

Any detection we do about container image paths should likely be done from the container manifest itself.

Reply:
I did this because the languages might be versioned as well (as they are in Python, and may be in Java). This version may not align precisely with the version used during construction, but the default container should always be compatible with it to whatever level of granularity our containers are build. I also don't think we should be encourage any branching on `beam:sdk:version`. However, if you feel strongly we could split these out. 

Adding a generic marker for go.

Reply:
The language version is an interesting point and brings up an issue where you could be using the same base container but the user submitting the pipeline is using a different version.

I just feel as though people will attempt to "parse" the language and version out of the container URL but if it comes to that we can always add more capabilities and remove existing ones.

Reply:
Yeah, we can't rule out people trying to parse these, but at least then they'll know they're on shaky ground. And as you say, these aren't terribly sticky. (I hope we never have to use them.)